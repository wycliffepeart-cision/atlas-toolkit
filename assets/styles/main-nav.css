@layer atlas {
  atlas-main-nav {
    align-items: center;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    margin-block-start: 5em;

    flex-direction: column;
    row-gap: 2em;
  }

  atlas-main-nav > atlas-main-nav-item > atlas-button {
    --atlas-button-primary-color: var(--atlas-color-primary-n300);
    --button-prefix-color: var(--atlas-color-primary-b400);

    --atlas-button-primary-background-color: transparent;
    --atlas-button-primary-disabled-background-color: transparent;
    --atlas-button-hover-background-color: transparent;
    --primary-active-background: transparent;

    --large-padding-top-bottom: 0;
    --large-padding-left-right: 0;
    --text-space-between: 0.5em;
  }

  atlas-main-nav > atlas-main-nav-item > atlas-button:hover {
    --button-prefix-color: var(--atlas-color-primary-o500);
  }

  atlas-main-nav > atlas-main-nav-item > atlas-button[selected]:not([selected='false']) {
    --atlas-button-primary-color: var(--atlas-color-primary-o500);
    --button-prefix-color: var(--atlas-color-primary-o500);
  }

  atlas-main-nav > atlas-main-nav-item > atlas-button {
    border: none;
  }

  atlas-main-nav > atlas-main-nav-item > atlas-button::part(prefix) {
    color: var(--button-prefix-color);
  }

  atlas-main-nav > atlas-main-nav-item > atlas-button::part(label) {
    display: none;
    font-size: var(--atlas-type-body-small-regular-font-size);
  }

  atlas-main-nav > atlas-main-nav-item > atlas-button > atlas-icon {
    color: inherit;
  }
}
