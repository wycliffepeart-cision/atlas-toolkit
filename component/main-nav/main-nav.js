import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var MainNavTemplate = "<template>\n  <slot></slot>\n</template>\n";

let MainNav = class MainNav extends HTMLElement {
};
MainNav = __decorate([
    Template(MainNavTemplate)
], MainNav);
if (!customElements.get('atlas-main-nav'))
    customElements.define('atlas-main-nav', MainNav);

export { MainNav };
