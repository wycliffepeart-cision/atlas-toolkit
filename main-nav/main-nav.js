import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var MainNavTemplate = "<template>\n  <slot></slot>\n</template>\n";

let MainNav = class MainNav extends HTMLElement {
};
MainNav = __decorate([
    Template(MainNavTemplate)
], MainNav);

export { MainNav };
