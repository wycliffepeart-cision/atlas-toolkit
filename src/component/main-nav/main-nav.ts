import MainNavTemplate from './main-nav.html';
import { Template } from '../../decorators/template';

@Template(MainNavTemplate)
export class MainNav extends HTMLElement {}

if (!customElements.get('atlas-main-nav')) customElements.define('atlas-main-nav', MainNav);
