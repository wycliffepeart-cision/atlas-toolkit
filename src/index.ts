export * from './component/accordion/accordion';
export * from './component/alert/alert';
export * from './component/application-frame/application-frame';
export * from './component/button/button';
export * from './component/icon/icon';
export * from './component/button-set/button-set';
export * from './component/chip/chip';
export * from './component/chip-group/chip-group';
export * from './component/dialog/dialog';
export * from './component/feed/feed';
export * from './component/input-checkbox/input-checkbox';
export * from './component/input-chip-group/input-chip-group';
export * from './component/input-dropdown/input-dropdown';
export * from './component/input-search/input-search';
export * from './component/input-text/input-text';
export * from './component/input-textarea/input-textarea';
export * from './component/logo/logo';
export * from './component/main-nav/main-nav';
export * from './component/main-nav-item/main-nav-item';
export * from './component/number-display/number-display';
export * from './component/profile-control/profile-control';
export * from './component/snackbar/snackbar';
export * from './component/stepper/stepper';
export * from './component/stepper-step/stepper-step';
export * from './component/tile-board/tile-board';
export * from './component/tile-board-tile/tile-board-tile';
export * from './component/workflow/workflow';
export * from './component/workflow-step/workflow-step';
