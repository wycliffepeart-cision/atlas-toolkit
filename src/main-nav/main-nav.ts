import MainNavTemplate from './main-nav.html';
import { Template } from '../decorators/template';

@Template(MainNavTemplate)
export class MainNav extends HTMLElement {}
