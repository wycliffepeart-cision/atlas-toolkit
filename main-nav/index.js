import { MainNav } from './main-nav.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-main-nav'))
    customElements.define('atlas-main-nav', MainNav);
