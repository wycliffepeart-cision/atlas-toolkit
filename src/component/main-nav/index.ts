import { MainNav } from './main-nav';

if (!customElements.get('atlas-main-nav')) customElements.define('atlas-main-nav', MainNav);
