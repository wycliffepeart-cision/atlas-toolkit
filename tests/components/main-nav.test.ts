import '../../src/main-nav';

const MAIN_NAV_TAG = 'atlas-main-nav';
const getShadowDomContent = (element) => element.shadowRoot.innerHTML;
const shadowContentSnapshot = `
	<template>
	</template>
`;

describe('MainNav', () => {
  it('is correctly appended in the DOM', async () => {
    await customElements.whenDefined(MAIN_NAV_TAG);
    const element = document.createElement(MAIN_NAV_TAG);
    document.body.appendChild(element);

    (expect(element) as any).toBeInTheDocument();
  });

  it('is capable of being selected', async () => {
    await customElements.whenDefined(MAIN_NAV_TAG);
    const element = document.createElement(MAIN_NAV_TAG);
    document.body.appendChild(element);

    const selectedElement = document.querySelector(MAIN_NAV_TAG);
    (expect(selectedElement) as any).toBeInTheDocument();
  });

  it('has the correct content when created', async () => {
    await customElements.whenDefined(MAIN_NAV_TAG);
    const element = document.createElement(MAIN_NAV_TAG);
    document.body.appendChild(element);

    (expect(element) as any).toContainHTML('<atlas-main-nav></atlas-main-nav>');
    (expect(element) as any).toBeEmptyDOMElement();
    (expect(getShadowDomContent(element)) as any).toMatchSnapshot(shadowContentSnapshot);
  });
});
