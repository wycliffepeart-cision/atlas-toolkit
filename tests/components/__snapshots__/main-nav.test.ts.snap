// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav has the correct content when created: 
	<template>
	</template>
 1`] = `
<slot>
</slot>
`;
